_posts/2022-05-17-data-location-inheritance-bug.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+published: true
+title: 'Bug Concerning Data Location during Inheritance'
+date: '2022-05-17'
+author: Solidity Team
+category: Security Alerts
+---
+
+On February 5th 2021, Nicolas Venturo reported a bug that allows
+overriding functions to change the data location of parameters from
+``memory`` to ``calldata``.
+
+TODO explain why it took us over a year to fix it.
+
+The bug was introduced in Solidity 0.6.9 together with the ability to use ``calldata``
+data location for all variables (and not just parameters of external functions).
+
+We assigned the bug a severity of "low".
+
+## Which Contracts are Affected?
+
+The effect of the bug is that a memory pointer is interpreted as a calldata pointer
+or vice-versa. It can only happen if you change the data location of a function
+during inheritance and perform a call at a location that is only aware of
+the function in the base contract (i.e. inside the base contract itself
```

Comment:
I'd clarify it like this, though I'm not sure if "signature" is 100% correct here since the `calldata`/`memory` is not part of it. Not sure how to call the whole declaration that includes it though.
```suggestion
the original function signature in the base contract (i.e. inside the base contract itself
```